Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -157,28 +181,45 @@ public GetObjectListArgs WithContinuationToken(string token)
             return this;
         }
 
+        public GetObjectListArgs WithListObjectsV1(bool useV1)
+        {
+            this.UseV2 = !useV1;
+            return this;
+        }
+
         internal override RestRequest BuildRequest(RestRequest request)
         {
             if (this.Versions)
             {
                 request.AddQueryParameter("versions", "");
             }
-            else
+            else if(this.UseV2)
             {
                 request.AddQueryParameter("list-type", "2");
             }
             request.AddQueryParameter("delimiter",this.Delimiter);
-            request.AddQueryParameter("prefix",this.Prefix);
+            if (!string.IsNullOrWhiteSpace(this.Prefix))
+            {
+                request.AddQueryParameter("prefix",this.Prefix);
+            }
             request.AddQueryParameter("max-keys", "1000");
             request.AddQueryParameter("encoding-type","url");
-            if (!string.IsNullOrWhiteSpace(this.KeyMarker))
+            if (!string.IsNullOrWhiteSpace(this.KeyMarker) && this.Versions)
             {
                 request.AddQueryParameter("key-marker",this.KeyMarker);
             }
-            if (!string.IsNullOrWhiteSpace(this.VersionIdMarker))
+            if (!string.IsNullOrWhiteSpace(this.VersionIdMarker) && this.Versions)
             {
                 request.AddQueryParameter("version-id-marker",this.VersionIdMarker);
             }
+            if (!string.IsNullOrWhiteSpace(this.Marker))
+            {
+                request.AddQueryParameter((this.UseV2)?"start-after":"marker",this.Marker);
+            }
+            if (!this.UseV2)
```

Comment:
isn't this check redundant? as it is already checked in the previous line?

Reply:
Correction made.
We need an 'empty' marker if 'max-keys' is defined and the result is marked 'IsTruncated'. Else in the response <Marker> field isn't filled leading to an erroneous request-response situation.

Reply:
* '<Marker>' in the response field isn't filled